src/jit/lsra.cpp

```diff
@@ -3596,7 +3602,13 @@ void LinearScan::buildRefPositionsForNode(GenTree*                  tree,
     int consume = info.srcCount;
     int produce = info.dstCount;
 
+#ifndef ARM_SOFTFP
     assert(((consume == 0) && (produce == 0)) || (ComputeAvailableSrcCount(tree) == consume));
+#else
+    assert(((consume == 0) && (produce == 0)) || ((consume == 1) && (produce == 2) && tree->OperGet() == GT_COPY &&
```

Comment:
This is not correct. It should always be the case that either the node consumes and produces no registers, or the latter condition holds. If this doesn't hold, then you have the wrong count on one of the nodes. It looks like the code below sets the `srcCount` to 1 and the `dstCount` to 2 for this case of `GT_COPY`, so why is this special case needed?